workspaces/orchestrator/entities/convertWorkflowToTemplate.yaml

```diff
@@ -0,0 +1,118 @@
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: convert-workflow-to-template
+  title: Convert serverless workflow to a template
+  description: Create a software template out of an existing serverless workflow.
+  tags:
+    - orchestrator
+
+spec:
+  owner: red-hat-developer-hub-authors
+  system: red-hat-developer-hub
+  type: service
+
+  # matches workflow's dataInputSchema
+  parameters:
+    - title: Provide inputs
+      required:
+        - workflow_id
+      properties:
+        workflow_id:
+          title: Workflow ID
+          type: string
+          ui:autofocus: true
+
+    - title: Choose a location
+      required:
+        - repoUrl
+      properties:
+        repoUrl:
+          title: Repository Location
+          type: string
+          ui:field: RepoUrlPicker
+          ui:options:
+            allowedHosts:
+              - github.com
+        pluginLocation:
+          type: string
+          title: Where to publish this new plugin?
+          ui:widget: radio
+          default: Create a new repository within the specified organization
+          enum:
+            - Submit a pull request to the same repository
+            - Create a new repository within the specified organization
+      dependencies:
+        pluginLocation:
+          allOf:
+            - if:
+                properties:
+                  pluginLocation:
+                    const: Submit a pull request to the same repository
+              then:
+                properties:
+                  branchName:
+                    title: Branch Name
+                    type: string
+                    description: The name for the branch
+                    default: ''
+                  targetBranchName:
+                    title: Target Branch Name
+                    type: string
+                    description: The target branch name of the merge request
+                    default: ''
+                required:
+                  - branchName
+                  - targetBranchName
+  steps:
+    - id: workflow_params
+      name: Retrieve workflow parameters
+      action: orchestrator:workflow:get_params
+      input:
+        workflow_id: ${{ parameters.workflow_id }}
+        indent: 4 # conforms indentation of the template yaml for the "parameters" property
+
+    - id: template
+      name: Fetch Skeleton + Template
+      action: fetch:template
+      input:
+        url: ./skeleton
+        values:
+          workflow_id: ${{ parameters.workflow_id }}
+          title: ${{ steps.workflow_params.output.title }}
```

Comment:
In your opinion is it possible that `steps.workflow_params.output` is null? What if `workflow_params` fail?
this doubt applies to other output variables too

Reply:
It would mean that even the `action: orchestrator:workflow:get_params` in previous step failed and we will receive an error anyway.

These values are provided programmatically, not by user. If they are missing, it's a bug to be handled in the `orchestrator:workflow:get_params` custom action.